terraform/20-app/main.tf

```diff
@@ -28,3 +35,53 @@ terraform {
     key            = "app/state.tfstate"
   }
 }
+
+module "cognito" {
```

Comment:
I appreciate this PR is just to communicate thoughts, so I'll just leave this as a reminder.

Nitpick: I think these application resources should go into a seperate place. Not in main.tf, that is kinda reserved for defining the backend state and the providers.
So in this case there would be a new file at `terraform/20-app/cognito.tf` containing this instantiation of the module you've made

Reply:
Completely agree, I have moved the cognito code to a separate `cognito.tf` file